### DISCUSSION TITLE
Question: Can UWP ComboBox max item display limit be overridden?

### DISCUSSION THREAD

**Discussion by YoungFcode** [microsoft/microsoft-ui-xaml#7503] [Q&A]
I'm using the UWP ComboBox (VS 2019). There is apparently a built in max display limit of approximately 9 items depending on item height. So for example, I have a ComboBox with 12 items, when initially opened (with a mouse click) it displays 9 of the items even though there is plenty of real estate to display all 12 items. 
If ComboBox is opened on a touch screen with a finger touch it displays ~6 items, even though there is plenty of real estate to display all 12 items. 
Notes: 
   - You can scroll through list to see all items.
   - MaxDropDownHeight = Default (Infinity)

Is there a way to override this behavior and just have all items display when ComboBox opened?

**Comment by StephenLPeters** [comment]
I think there is not anything exposed that would allow you to fix this.. @MikeHillberg maybe you know some secrets?

**Comment by YoungFcode** [comment]
Moving forward is there any chance Microsoft will be updating the UWP ComboBox to allow display of all items?

**Comment by StephenLPeters** [comment]
ComboBox is not a control shipped in Winui2, meaning changing it would require an OS update or Winui3.  We could examine this once WinUI3 ships.

**Comment by YoungFcode** [comment]
I would highly recommend this update.

**Comment by licanhua** [comment]
I didn't do the testing, maybe you can try re-template the ComboBox and set MaxDropDownHeight to a larger number. Currently it's hard coded with 504

 <Setter Property="MaxDropDownHeight" Value="504" />

https://github.com/microsoft/microsoft-ui-xaml/blob/c464079156f3731165ba5068277935c8ed174207/dev/ComboBox/ComboBox_themeresources.xaml#L393

**Comment by YoungFcode** [comment]
In generic.xaml (v10.0.18362.0) I do see a hardcoded MaxDropDownHeight = "504" for the ComboBox. 
My project is actually using generic.xaml (v10.0.17763.0) which does not contain a MaxDropDownHeight (for any control).
Just to be sure I changed MaxDropDownHeight = "1200" in v10.0.18362.0, but had no effect.
I also explicitly set MaxDropDownHeight on the ComboBox control itself in the xaml, again to no effect.
This behaviour is apparently baked into the ComboBox control
